steampy/client.py

```diff
@@ -207,11 +209,14 @@ def _get_trade_offer_url(self, trade_offer_id: str) -> str:
         return self.COMMUNITY_URL + '/tradeoffer/' + trade_offer_id
 
     def _confirm_transaction(self, trade_offer_id: str) -> dict:
-        confirmation_executor = ConfirmationExecutor(trade_offer_id,
-                                                     self.steam_guard['identity_secret'],
-                                                     self.steam_guard['steamid'],
+        confirmation_executor = ConfirmationExecutor(self.steam_guard['identity_secret'], self.steam_guard['steamid'],
                                                      self._session)
-        return confirmation_executor.send_trade_allow_request()
+        return confirmation_executor.send_trade_allow_request(trade_offer_id)
+
+    def _confirm_sell_listing(self, asset_id: str) -> dict:
+        con_executor = ConfirmationExecutor(self.steam_guard['identity_secret'], self.steam_guard['steamid'],
+                                            self._session)
+        return con_executor.confirm_sell_listing(asset_id)
 
```

Comment:
I made the ConfirmatorExecutor more general, now there are two methods to confirm something (one for tradeoffers and the second for the sell_listings).
So instead of instanciate the object and pass the trade_offer_id, we first just instanciate the object in a more general way without passing any ids, then we call it on the right method passing it the right id.